pkg/render/typha.go

```diff
@@ -700,3 +703,31 @@ func (c *typhaComponent) affinity() (aff *corev1.Affinity) {
 	}
 	return aff
 }
+
+// typhaPrometheusService service for scraping typha metrics.
```

Comment:
Our docs currently include steps to create this service: https://docs.tigera.io/calico-enterprise/latest/operations/monitor/prometheus/byo-prometheus

Just adding a reminder here that we'll need to submit a corresponding docs PR to remove this step.

Reply:
Great catch. External prometheus and the current steps for byo prometheus individual settings are supposed to be mutually exclusive. And if they are used in combination then you are right that that service is no longer necessary. The docs changes can be found here: https://github.com/tigera/docs/pull/1192